lib/resty/core/request.lua

```diff
@@ -132,6 +147,80 @@ function ngx.req.get_headers(max_headers, raw)
 end
 
 
+function ngx.req.get_post_args(max_args)
+    local r = getfenv(0).__ngx_req
+    if not r then
+        return error("no request found")
+    end
+
+    if not max_args then
+        max_args = -1
+    end
+
+    local rc = C.ngx_http_lua_ffi_req_validate_post(r)
+
+    if rc == FFI_BAD_CONTEXT then
+        return error("API disabled in the current context")
+    end
+
+    if rc == FFI_POST_ARGS_EMPTY then
+        return {}
+    end
+
+    if rc == FFI_POST_ARGS_UNREAD then
+        return error("no request body found; maybe you should turn on lua_need_request_body?")
+    end
+
+    if rc == FFI_POST_ARGS_IN_FILE then
+        return nil, "request body in temp file not supported"
+    end
+
+    local args_len = C.ngx_http_lua_ffi_req_get_post_args_len(r)
+
+    if args_len == 0 then
+        return {}
+    end
+
+    local strbuf = get_string_buf(args_len)
+    local kvbuf = ffi_cast(table_elt_type, strbuf + args_len)
+
+    local n = C.ngx_http_lua_ffi_req_get_post_args_count(r, strbuf,
+        args_len, max_args)
+
+    local nargs = C.ngx_http_lua_ffi_req_get_post_args(r, strbuf,
+        kvbuf, n, args_len)
+
+    local args = new_tab(0, nargs)
+    for i = 0, nargs - 1 do
+        local arg = kvbuf[i]
```

Comment:
Should we go against Mike Pall's advice not to cache table lookup in such a hot code path? I am not sure, just to be clear, just raising a concern we previously talked about :)

This happens a bunch in this function, although, I do agree that it makes the code more readable :)